jerry-core/include/jerryscript-core.h

```diff
@@ -159,6 +159,18 @@ typedef struct
 } jerry_property_descriptor_t;
 
 /**
+ * Description of JerryScript heap memory stats.
+ * It is for memory profiling.
+ */
+typedef struct
+{
+  size_t version; /**< the version of the stats struct */
```

Comment:
@zherczeg @LaszloLango @fbmrk @martijnthe 
What about 
```
typedef struct
{
  size_t size; /**< heap total size */
  size_t allocated_bytes; /**< currently allocated bytes */
  size_t peak_allocated_bytes; /**< peak allocated bytes */
  size_t waste_bytes; /**< bytes waste due to blocks filled partially */
  size_t peak_waste_bytes;
}  jerry_heap_stats_t
```
I think those fields are not much impl-dependent comparing to `alloc_count`, `alloc_iter_count`.
And maybe we can add `byte_code_bytes`, `string_bytes`, and so on.